sdk/storage/azdatalake/file/client.go

```diff
@@ -483,6 +490,10 @@ func (f *Client) uploadFromReader(ctx context.Context, reader io.ReaderAt, actua
 	})
 
 	if err != nil {
+		_, err2 := f.Delete(ctx, nil)
```

Comment:
Should this code be moved after FlushData operation?

Reply:
Since, the file is created if `o.EncryptionContext` is not nil. Should we also add this check before deleting the file?

Reply:
As discussed offline, 
For 1. Flush is called only when upload is successful, and we are checking upload errors itself. So, better if its not moved after FlushData operation.
For 2. Yes, added a check for the same before deleting the file.